### ISSUE TITLE
date property loses timezone on change

### ISSUE THREAD

**Issue by georgpfolz** [zopefoundation/Zope#1144]
When I create a date property on an object and assign it a DateTime, it stores the datetime string including the timezone, but when I update the property, the timezone information is lost

```
date = context.ZopeTime('4.5.2023 13:45')
print(date)   # prints:  2023/04/05 13:45:00 GMT+2

my_obj.manage_addProperty('test_prop', date, 'date')  # stores 2023/04/05 13:45:00 GMT+2
my_obj.manage_changeProperties('test_prop', date)  # stores 2023/04/05 13:45:00, the timezone is gone
```

As a workaround, I can do `my_obj.manage_changeProperties('test_prop', str(date)`, but I would have expected that a **DateTime** saved in a date field would be saved the same whether I create or update the property.

**Comment by d-maurer** [comment]
You call `manage_changeProperties` wrongly: its signature is `self, REQUEST=None, **kw`, i.e. the correct call would be `manage_changeProperties(test_prop=date)`.

I checked this in the current `Zope` master:
```pycon
>>> date=app.ZopeTime('4.5.2023 13:45')
>>> date
DateTime('2023/04/05 13:45:00 GMT+2')
>>> app.manage_addProperty('test_prop', date, 'date')
>>> app.getProperty("test_prop")
DateTime('2023/04/05 13:45:00 GMT+2')
>>> app.manage_changeProperties(test_prop=date)
>>> app.getProperty("test_prop")
DateTime('2023/04/05 13:45:00 GMT+2')
```

If you look at the code (--> `OFS.PropertyManager`), you will see that the property value is simply stored if it is not a string. Thus, a timezope remains unchanged.

**Comment by georgpfolz** [comment]
> You call manage_changeProperties wrongly

Sorry, I made an error in writing my post. I fact I did call it correctly:
`my_obj.manage_changeProperties({'test_prop': date})`
(the wrong call produces would produce a TypeError anyway).

I checked the code in PropertyManager: in both cases, the value is still a DateTime object when it's set in **_setPropValue**, so I did what I should have done at the start:

```
my_obj.manage_changeProperties(test_prop=date)
print(my_obj.getProperty('test_prop')  #  prints the date with the timezone
```

So in the end, this is not really an issue, the representation in the ZMI ist just somewhat confusing. I'm not entirely sure what's going on: first I thought that the timezone information is omitted if it's the local timezone, but if I create the date object like this `DateTime(''4.5.2023 13:46 GMT+2')`, the timezone is always shown.

I checked: the timezone information also does not get lost if I save the date field in the ZMI (or maybe it's just reset to the local timezone every time)
